### PR TITLE
Support configurable Allowed Content Types within Hypertrace Javaagent

### PR DESCRIPTION
**Pull request by bradAtTraceable**
I had to start with a new PR and branch on this issue, due to the problems I was having pushing to the origin branch.  This, I believe, was due to submodules within the Hypertrace agent.  I think that these problems are fixed with this PR.

### CODE REVIEW

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java [hunk 1]
```diff
@@ -189,4 +219,27 @@ private static String convertYamlToJson(InputStream yaml) throws IOException {
     ObjectMapper jsonWriter = new ObjectMapper();
     return jsonWriter.writeValueAsString(obj);
   }
+
+  /**
+   * Creates a collection of objects consisting of the set of objects specified in the
```
Comment (tim-mwangi) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java:224]: Should we have the config allowed content types just do the override? i.e if you set the config yourself, you need to add the defaults as well in the config i.e

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/DataCaptureConfigImpl.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.google.protobuf.StringValue;
+import java.util.List;
+import org.hypertrace.agent.config.v1.Config;
+import org.hypertrace.agent.core.config.DataCaptureConfig;
+
+/**
+ * An implementation of the DataCaptureConfig interface, which can provide the names of the content
+ * types that can be captured.
+ */
+@AutoService(DataCaptureConfig.class)
```
Comment (tim-mwangi) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/DataCaptureConfigImpl.java:29]: Why can't the default allowed content types be just a list of strings?

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java (hunk 1 above)
Comment (ryanericson) [reply otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java:224]: yep

Comment (bradAtTraceable) [reply otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java:224]: ok, I wasn't sure what was the right thing to do here.  Will fix

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/DataCaptureConfigImpl.java (hunk 2 above)
Comment (bradAtTraceable) [reply otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/DataCaptureConfigImpl.java:29]: I followed the precedent of defining implementations for the portions of the configuration that are required by the Java agent.  The Java agent uses Java services to perform the initialization of config data; I merely continued the practice, for consistency.

Comment (tim-mwangi) [reply otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/DataCaptureConfigImpl.java:29]: I see. Ok.

File: otel-extensions/src/test/resources/nonDefaultDataCaptureConfig.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  trace_reporter_type: OTLP
+  metric_reporter_type: METRIC_REPORTER_TYPE_OTLP
+  cert_file: /foo/bar/example.pem
+dataCapture:
+  bodyMaxSizeBytes: 16
+  httpHeaders:
+    request: true
+    response: false
+  httpBody:
+  allowed_content_types: ["foo", "bar"]
```
Comment (shashank11p) [inline otel-extensions/src/test/resources/nonDefaultDataCaptureConfig.yaml:19]: Should we only use the ones given here or add these along with the default? I guess using only the ones given in config would give us options to remove some content-types if we do not want to capture those.

Comment (tim-mwangi) [reply otel-extensions/src/test/resources/nonDefaultDataCaptureConfig.yaml:19]: https://github.com/hypertrace/javaagent/pull/367#discussion_r946068547

Comment (tim-mwangi) [reply otel-extensions/src/test/resources/nonDefaultDataCaptureConfig.yaml:19]: We'd like the config to override the defaults. So if you set the config then only those in the config will be used.

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**